xml-conduit/test/unit.hs

```diff
@@ -740,6 +741,14 @@ resolvedInline = do
     Res.Document _ root2 _ <- return $ Res.parseLBS_ Res.def "<!DOCTYPE foo [<!ENTITY bar \"baz\">]><foo bar='&bar;'/>"
     root2 @?= Res.Element "foo" (Map.singleton "bar" "baz") []
 
+resolvedInlineComplex :: Assertion
+resolvedInlineComplex = do
+    Res.Document _ root _ <- return $ Res.parseLBS_ Res.def "<!DOCTYPE foo [<!ENTITY bar \"<p>baz &bim;</p>\"><!ENTITY bim \"Hello\">]><foo>&bar;</foo>"
+    root @?= Res.Element "foo" Map.empty [Res.NodeElement (Res.Element "p" Map.empty [Res.NodeContent "baz Hello"])]
+    Res.Document _ root2 _ <- return $ Res.parseLBS_ Res.def "<!DOCTYPE foo [<!ENTITY bar \"<p>baz</p>\">]><foo class=\"&bar;\"/>"
```

Comment:
Shouln't this expand to `<foo class="<p>baz</p>"/>`, and thus be invalid as [character `<` is forbidden in attribute value](https://www.w3.org/TR/REC-xml/#NT-AttValue) ?

Reply:
I wasn't sure how to handle this.  The current code simply ignores non-textual content when expanding entities inside attributes.  Attributes are `(TName, [Content])`, so there's no way to include non-textual content with the current types.

I'm also not really sure how XML wants you to handle this case.

Reply:
I think you're right that XML wants you to expand this into an invalid document.  But it would be hard to do that without bigger changes to the types and functions.  We can't even raise an exception in the function that handles this (`tokenToEvent`).  We could leave the ContentEntity as it is, which would trigger an unresolved entity exception later, but this would be a pretty puzzling error to get when the entity is defined. A longer-term solution would be to add a parameter to ContentEntity that allows us to track why the entity wasn't expanded. This would allow us to raise a more informative exception later.  Or the plumbing could be rearranged so that an exception could be raised from `tokenToEvent`.